staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -900,11 +900,19 @@ func (c *Cacher) processEvent(event *watchCacheEvent) {
 	c.incoming <- *event
 }
 
-func (c *Cacher) dispatchEvents() {
+func (c *Cacher) dispatchEvents(progressRequester *conditionalProgressRequester) {
 	// Jitter to help level out any aggregate load.
 	bookmarkTimer := c.clock.NewTimer(wait.Jitter(time.Second, 0.25))
 	defer bookmarkTimer.Stop()
 
+	// TODO(p0lyn0mial): adapt the following logic once
+	//  https://github.com/kubernetes/kubernetes/pull/124612 merges
+	progressRequesterCleanUpOnceFn := func() { /*no-op*/ }
+	if utilfeature.DefaultFeatureGate.Enabled(features.ConsistentListFromCache) {
+		progressRequester.Add()
+		progressRequesterCleanUpOnceFn = sync.OnceFunc(progressRequester.Remove)
+	}
+
```

Comment:
So with all discussion on the other PRs, I think it will actually be much easier to proceed with a different approach (the one that I thought about originally).

Basically, instead of even requesting bookmarks, ensure that we initialize `lastProcessedResourceVersion` correctly from the beginning.
So the flow we want to achieve is that `lastProcessedResourceVersion` will actually be set when the first List call is done.

Now, synchronizing that correctly in arbitrary way is a bit tricky, but we have a pretty simple path to achieve it.

What you just need to do is very simple thing, just add here:
```
lastProcessedResourceVersion := uint64(0)
PollUntil(10ms, func() (bool, error) {
  if rv := c.watchCache.GetResourceVersion(); rv != 0 {
    lastProcessedResourceVersion = rv
    return true, nil
  }
  return false, nil
}
```

That solves the whole problem - you don't need any other changes in this PR.

Reply:
ok, i like the idea it will be detached from the `progressRequester`. thanks!